connectors/sync_job_runner.py

```diff
@@ -143,20 +167,28 @@ async def _sync_done(self, sync_status, sync_error=None):
             "indexed_document_count": result.get("indexed_document_count", 0),
             "indexed_document_volume": result.get("indexed_document_volume", 0),
             "deleted_document_count": result.get("deleted_document_count", 0),
-            "total_document_count": await self.connector.document_count(),
         }
 
-        if sync_status == JobStatus.ERROR:
-            await self.sync_job.fail(sync_error, ingestion_stats=ingestion_stats)
-        elif sync_status == JobStatus.SUSPENDED:
-            await self.sync_job.suspend(ingestion_stats=ingestion_stats)
-        elif sync_status == JobStatus.CANCELED:
-            await self.sync_job.cancel(ingestion_stats=ingestion_stats)
-        else:
-            await self.sync_job.done(ingestion_stats=ingestion_stats)
+        if await self.reload_sync_job():
+            if await self.reload_connector():
+                ingestion_stats[
+                    "total_document_count"
+                ] = await self.connector.document_count()
+
+            if sync_status == JobStatus.ERROR:
+                await self.sync_job.fail(sync_error, ingestion_stats=ingestion_stats)
+            elif sync_status == JobStatus.SUSPENDED:
+                await self.sync_job.suspend(ingestion_stats=ingestion_stats)
+            elif sync_status == JobStatus.CANCELED:
+                await self.sync_job.cancel(ingestion_stats=ingestion_stats)
+            else:
+                await self.sync_job.done(ingestion_stats=ingestion_stats)
+
+            await self.reload_sync_job()
+
+        if await self.reload_connector():
+            await self.connector.sync_done(self.sync_job)
```

Comment:
Same, what about `else`?

Reply:
`else` means the connector is somehow deleted, so `self.connector.sync_done` will not be called. Will add a log for `else`.

Reply:
Same remark, I think it's worth to add a log line here